examples/config-stream-cache.json

```diff
@@ -0,0 +1,40 @@
+{
+  "distSpecVersion": "1.1.0",
+  "storage": {
+    "rootDirectory": "./tmp/zot"
+  },
+  "http": {
+    "address": "0.0.0.0",
+    "port": "8080"
+  },
+  "log": {
+    "level": "debug"
+  },
+  "extensions": {
+    "sync": {
+      "enable": true,
+      "registries": [
+        {
+          "urls": [
+            "https://registry.access.redhat.com"
+          ],
+          "onDemand": true,
+          "maxRetries": 3,
+          "retryDelay": "5m",
+          "tlsVerify": true,
+          "streamCache": {
```

Comment:
We already download blobs to a temporary dir, so no need to create a new one.
We likely just need "onDemandStreaming": enable/disable and which should depend on onDemand:true